### ISSUE TITLE
pass additional parameters to estimator .fit method

### ISSUE THREAD

**Issue by mangos47** [scikit-optimize/scikit-optimize#909]
I was using skopt.BayesSearchCV with sklearn wrapper of xgboost to tune hyperparameter. below is my code snippet:
`    clf = XGBClassifier(objective = 'binary:logistic',booster='gbtree',tree_method='approx',
                        n_jobs = int(n_cpu_use),verbosity =3,seed=42)
    opt = BayesSearchCV(clf,search_spaces,cv=skf,n_iter=150,scoring = my_scorer,
                        n_jobs=1,refit=True,optimizer_kwargs={'base_estimator': 'RF'},  
                        random_state=42,verbose=1)
    bys_result = opt.fit(X_train, y_train, callback=[status_print])
`
when I use an older version of xgboost, I can put early_stopping_rounds inside XGBClassifier and it seems it'll work (by gauging speed).
Now I upgraded to newer version of xgboost. early_stopping_rounds is not a parameter in XGBClassifier.fit method, not in creator method. But BayesSearchCV seems do not support pass additional parameter to estimator's fit method.
How can I use early_stopping for xgb with BayesSearchCV now?
thank you
